### ISSUE TITLE
MSA rendering is very resource intensive[For huge tree/Sequences] browser crashes

### ISSUE THREAD

**Issue by n3m3sis00** [n3m3sis00/phylo-react#42]
* TREE-MSA rendering for `ENSGT00390000003602` crashes the browser.
* Reason is MSA rendering, because of the functionality of coloring, each letter is processed separately.

FIX:
* One Fix might be to check for total length of all the sequences if it above the certain threshold render without color Sequences

**Comment by cmdcolin** [comment]
This might suggest the need to use something such as HTML5 canvas for the MSA also. Large sequences do indeed end up producing a lot of nodes

**Comment by n3m3sis00** [comment]
> This might suggest the need to use something such as HTML5 canvas for the MSA also. Large sequences do indeed end up producing a lot of nodes

Yes we can introduce html canvas, what I was thinking of using a GPU accelerated library(like Three.js) as well.

**Comment by cmdcolin** [comment]
Canvas should be fast enough and canvas is GPU accelerated in most browsers such as chrome. Canvas just doesn't use webgl, which three.js does

**Comment by cmdcolin** [comment]
Also, just to be clear about what causes the slowdown, it is large numbers of DOM nodes. Browsers cannot handle tens of thousands of DOM nodes basically. Canvas just "draw" rather than render to DOM nodes so this is why they are faster.

**Comment by n3m3sis00** [comment]
> what causes the slowdown, it is large numbers of DOM nodes.

Yes exactly

**Comment by n3m3sis00** [comment]
I Will try to port Current MSA component to Canvas

**Comment by cmdcolin** [comment]
Sounds good:)
